plugins/ingest/src/test/java/org/elasticsearch/ingest/processor/conditional/ConditionalProcessorFactoryTests.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Licensed to ElasticSearch and Shay Banon under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. ElasticSearch licenses this
+ * file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.ingest.processor.conditional;
+
+import org.elasticsearch.ingest.processor.Processor;
+import org.elasticsearch.test.ESTestCase;
+import org.junit.Before;
+
+import java.util.*;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.mockito.Matchers.anyMap;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class ConditionalProcessorFactoryTests extends ESTestCase {
+    ConditionalProcessor.Factory factory;
+    Map<String, Object> config;
+    List<Map<String, Object>> matchedProcessors;
+    List<Map<String, Object>> unmatchedProcessors;
+    Processor anyProcessor;
+
+    @Before
+    public void init() throws Exception {
```

Comment:
maybe also test a nested conditional setup? (So have conditional and then another conditional in the matched or unmatched list)